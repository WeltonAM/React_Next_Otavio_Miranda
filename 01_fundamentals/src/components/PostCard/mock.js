export const postCardPropsMock = {
    post: [],
    cover: 'img/img.png',
    title: 'title 1',
    body: 'body 1',
    id: 1
}