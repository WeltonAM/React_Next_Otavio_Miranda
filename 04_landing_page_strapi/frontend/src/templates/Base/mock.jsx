import linksMock from '../../components/NavLinks/mock';
import gridMock from '../../components/GridText/mock';

import GridText from '../../components/GridText/GridText';

const mockBase = {
    children: (
        <>
            <GridText {...gridMock} background />
            <GridText {...gridMock} />
            <GridText {...gridMock} background />
            <GridText {...gridMock} />
            <GridText {...gridMock} background />
            <GridText {...gridMock} />
        </>
    ),

    links: linksMock,

    logoData: {
        text: 'Logo',
        link: '#',
    },

    footerHtml: '<p>Footer test</p>',
};

export default mockBase;