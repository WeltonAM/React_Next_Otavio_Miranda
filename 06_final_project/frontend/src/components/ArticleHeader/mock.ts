import { ArticleHeaderProps } from '.';
import { data } from '../../api/data.json';

const title = data.posts.data[0].attributes.title;
const excerpt = data.posts.data[0].attributes.excerpt;
const cover = data.posts.data[0].attributes.cover.data.attributes;
const author = data.posts.data[0].attributes.author.data.attributes.displayName;
const categories = data.posts.data[0].attributes.categories.data[1].attributes.displayName;
const createdAt = data.posts.data[0].attributes.createdAt;

console.log(categories);

export default {
    title,
    excerpt,
    cover,
    author,
    categories,
    createdAt,
} as unknown as ArticleHeaderProps;