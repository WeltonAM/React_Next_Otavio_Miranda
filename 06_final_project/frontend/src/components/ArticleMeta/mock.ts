import { ArticleMetaProps } from '.';
import { data } from '../../api/data.json';

const createdAt = data.posts.data[0].attributes.createdAt;
const author = data.posts.data[0].attributes.author.data;
const categories = data.posts.data[0].attributes.categories.data;

export default {
    createdAt,
    author: {
        id: author.id,
        slug: author.attributes.slug,
        displayName: author.attributes.displayName,
    },
    categories: [
        {
            id: categories[0].id,
            displayName: categories[0].attributes.displayName,
            slug: categories[0].attributes.slug,
        },
        {
            id: categories[1].id,
            displayName: categories[1].attributes.displayName,
            slug: categories[1].attributes.slug,
        },
    ],
} as ArticleMetaProps;