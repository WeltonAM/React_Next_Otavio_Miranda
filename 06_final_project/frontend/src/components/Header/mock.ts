import { HeaderProps } from '.';
import { data } from '../../api/data.json';

const blogName = data.setting.data.attributes.blogName;
const blogDescription = data.setting.data.attributes.blogDescription;
const logo = data.setting.data.attributes.logo.data.attributes.url;

export default {
    blogName,
    blogDescription,
    logo,
    showText: true,
} as HeaderProps;