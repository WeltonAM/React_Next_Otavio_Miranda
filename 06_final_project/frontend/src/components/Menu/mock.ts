import { MenuProps } from '.';
import { data } from '../../api/data.json';

const links = data.setting.data.attributes.menuLink;
const blogName = data.setting.data.attributes.blogName
const logo = data.setting.data.attributes.logo.data.attributes.url;

export default {
    links,
    blogName,
    logo,
} as MenuProps;