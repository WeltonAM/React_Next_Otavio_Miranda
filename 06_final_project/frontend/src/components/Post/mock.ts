import { PostProps } from '.';
import { data } from '../../api/data.json';

const title = data.posts.data[0].attributes.title;
const excerpt = data.posts.data[0].attributes.excerpt;
const cover = data.posts.data[0].attributes.cover.data.attributes;
const content = data.posts.data[0].attributes.content;
const author = data.posts.data[0].attributes.author.data.attributes;
const categories = data.posts.data[0].attributes.categories.data;
const createdAt = data.posts.data[0].attributes.createdAt;

export default {
    title,
    excerpt,
    cover,
    content,
    author,
    categories,
    createdAt,
} as unknown as PostProps;