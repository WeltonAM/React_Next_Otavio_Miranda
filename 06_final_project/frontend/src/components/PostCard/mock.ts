import { PostCardProps } from '.';
import { data } from '../../api/data.json';

const title = data.posts.data[0].attributes.title;
const cover = data.posts.data[0].attributes.cover.data.attributes;
const excerpt = data.posts.data[0].attributes.excerpt;
const slug = data.posts.data[0].attributes.slug;

export default {
    title,
    excerpt,
    cover,
    slug
} as unknown as PostCardProps;