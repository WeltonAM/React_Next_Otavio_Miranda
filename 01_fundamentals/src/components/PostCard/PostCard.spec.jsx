/* eslint-disable testing-library/no-node-access */
import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const post = postCardPropsMock;

describe('<PostCard />', () => {
    it('should render PostCard correctly', () => {
        render(<PostCard {...post} />);
    
        expect(screen.getByAltText(/title 1/i)).toHaveAttribute('src', 'img/img.png');
        expect(screen.getByRole('heading', { title: 'title 1 1' })).toBeInTheDocument();
        expect(screen.getByText('body 1')).toBeInTheDocument();
      });
    
      it('should match snapshot', () => {
        const { container } = render(<PostCard {...post} />);
        expect(container.firstChild).toMatchSnapshot();
      });
});