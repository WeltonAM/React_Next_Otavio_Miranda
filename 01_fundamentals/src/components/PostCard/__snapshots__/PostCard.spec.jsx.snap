// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img />
  <div
    class="post-content"
  >
    <h3 />
    <p />
  </div>
</div>
`;
